posthog-core/src/eventemitter.ts

```diff
@@ -0,0 +1,27 @@
+export class SimpleEventEmitter {
+  events: { [key: string]: ((e: any) => void)[] } = {}
+
+  constructor() {
```

Comment:
Consider introducing/adding a `shutdown` or similar named function which can be used to clean up of things like:

- Call `flush` so things get flushed
- Stop any running timers, e.g. setInterval / setTimeout etc  
- Ensure `unref()` is called on timers, see also: https://httptoolkit.tech/blog/unblocking-node-with-unref/
- Abort  any on-going requests (if that makes sense?)

Reply:
Yep! Sounds good and what I'm hoping todo (I want separate tests for the nodejs side of things as things like unref are unique to that platform